### PR TITLE
Improving initialization, adding retries to auth and data broker requests.

### PR DESCRIPTION
**Pull request by giovannicuriel**
This PR adds three changes:

- Now Producer initialization is performed within Messenger's `init` function. This way a user can identify whether its initial connection succeeded or not.
- Auth requests will be sent a few times (set by conf.auth.connectionRetries) before generating an error (which can be detected by the user)
- The same thing for Data Broker requests.

This is connected to dojot/dojot#781 and is connected to dojot/dojot#888.

### CODE REVIEW

File: lib/auth.js [hunk 1]
```diff
@@ -21,18 +23,32 @@ function getManagementToken(tenant) {
   );
 }
 
-async function getTenants(auth) {
+function getTenants(auth, counter = 3) {
```
Comment (giovannicuriel) [inline lib/auth.js]: I didn't want to change this call.

File: lib/auth.js [hunk 2]
```diff
@@ -21,18 +23,32 @@ function getManagementToken(tenant) {
...
+      logger.error(`Tenants retrieved: ${response.data.tenants}.`, TAG);
+      resolve(response.data.tenants);
+    }).catch((error) => {
+      logger.error(`Could not initialize tenancy consumer: ${error}.`, TAG);
+      logger.debug(`Trying again in a few moments.`, TAG);
+      counter--;
+      logger.debug(`Remaining ${counter} time(s).`, TAG);
+      if (counter > 0) {
+        setTimeout(() => {
+          getTenants(auth);
```
Comment (ericbaum) [inline lib/auth.js]: Shouldn't the counter be decremented and passed to the function here?

File: lib/kafka/TopicManager.js [hunk 3]
```diff
@@ -2,6 +2,10 @@
 
 var axios = require("axios");
 var auth = require("../auth");
+var config = require("../config");
+var logger = require("@dojot/dojot-module-logger").logger;
+
+const TAG = { filenane: "topic-mgr" };
```
Comment (ericbaum) [inline lib/kafka/TopicManager.js]: filename

File: lib/auth.js (hunk 2 above)
Comment (giovannicuriel) [reply lib/auth.js]: Yes.

Comment (giovannicuriel) [reply lib/auth.js]: Same for TopicManager

File: lib/kafka/TopicManager.js [hunk 4]
```diff
@@ -24,16 +28,33 @@ class TopicManager {
 
     const querystring = global ? "?global=true" : "";
     const url = `${broker}/topic/${subject + querystring}`;
+    var counter = config.databroker.connectionRetries;
```
Comment (ericbaum) [inline lib/kafka/TopicManager.js]: Similar issue here as the counter will be reset within each execution

**Comment by giovannicuriel** [comment]
The retry counter for both Auth and TopicManager are wrong. I'll close this and create another PR with the fixes.

**Pull request closed without merging**